services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories.impl;
+
+import org.openkilda.model.Isl;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.repositories.IslRepository;
+
+import org.neo4j.ogm.cypher.query.SortOrder;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Neo4J OGM implementation of {@link IslRepository}.
+ */
+public class Neo4jIslRepository extends Neo4jGenericRepository<Isl> implements IslRepository {
+    public Neo4jIslRepository(Neo4jSessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public Isl findByEndpoint(SwitchId switchId, int port) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("src_switch", switchId.toString());
+        parameters.put("src_port", port);
+
+        return getSession().queryForObject(Isl.class, "MATCH (src:switch)-[target:isl]->(:switch)\n"
+                + " WHERE src.name=$src_switch AND target.src_port=$src_port\n"
+                + " RETURN target", parameters);
+    }
+
+    @Override
+    public Iterable<Isl> findOccupiedByFlow(String flowId, boolean ignoreBandwidth, long requiredBandwidth) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("flow_id", flowId);
+        parameters.put("ignore_bandwidth", ignoreBandwidth);
+        parameters.put("requested_bandwidth", requiredBandwidth);
+
+        String query = "MATCH (src:switch)-[fs:flow_segment{flowid: $flow_id}]->(dst:switch) "
+                + "MATCH (src)-[link:isl]->(dst) "
+                + "WHERE src.state = 'active' AND dst.state = 'active' AND link.status = 'active' "
```

Comment:
Why we force "active" status for ISL and switches, if this is "generic use function"? Why for we have a "magic" with ignore bandwidth and required bandwidth here?

Or method name or query is incorrect.

Reply:
Nice catch! I've renamed the method. Also I split it into 2 methods: one is with the requiredBandwidth argument and another which has no bandwidth requirement.